src/droplet.F

```diff
@@ -3216,20 +3247,33 @@ subroutine droplet_injection(pdata,xf,yf,time_since_inject,mtime,my_reintro)
 
       totdrops = dFssum(101)    !totdrops at this point is # per unit area per unit time
 
-      !Define these here
-      !These are the LOCAL area for EACH processor
+
+      !Define xrange and yrange to be WHOLE domain area, to calculate the total droplet number
+      xrange = dx*nx
+      yrange = dy*ny
+
+      my_reintro_tmp = xrange*yrange*time_since_inject*totdrops/drop_mult
+#ifdef MPI
+      my_reintro = floor(my_reintro_tmp/sngl(numprocs))
+
+      !Processors 0 through the remainder-1 will create an extra particle
+      if (myid .lt. floor(mod(my_reintro_tmp,sngl(numprocs)))) then 
+         my_reintro = my_reintro + 1
+      end if
```

Comment:
The number of new droplets to inject is being computed in `my_reintro_tmp`. An issue that made the actual number of new droplets in the code depend on number of MPI ranks is due to the neglected mod(my_reintro_tmp,numprocs). Rather than put all of these remaining particles onto a single rank, I'm having ranks 1:mod all inject one extra.

Reply:
Thanks for the clarification.